src/database/DataTypes/MasterList.js

```diff
@@ -44,18 +46,20 @@ export class MasterList extends Realm.Object {
   }
 
   /**
-   * Find the current stores matching store tag object in this master lists program settings
+   * Find the current stores matching store tag object in this master lists program settings.
+   * Program settings is a JSON object from 4D, which requires two parses.
    * @param  {string}  tags   Current stores tags field
    * @return {object} The matching storeTag programsettings field for the current store
    */
   getStoreTagObject(tags) {
+    if (!this.programSettings) return null;
     const storeTags = tags.split(/[\s,]+/);
-    return Object.entries(JSON.parse(this.programSettings).storeTags).reduce(
-      ([programStoreTag, storeTagObject]) => {
-        if (!(storeTags.indexOf(programStoreTag) >= 0)) return null;
-        return storeTagObject;
-      }
-    );
+    const firstParse = JSON.parse(this.programSettings);
+    if (!firstParse) return null;
+
+    return Object.entries(JSON.parse(firstParse).storeTags).filter(
```

Comment:
Would be safer is you check if storeTags exist. Also it's fine to break this down further so it's more readable, i.e;
* Use keys instead of entries 
* Then return foundKey ?  parsedProgramSettings[foundKey] : null

Reply:
done b983e71ac72281f8329e9de43dd87dbdf1ce6aac